get-i18n.js

```diff
@@ -6,25 +6,40 @@ const isBlock = line => /^[A-Z]/.test(line)
 const getBlock = line => line.split(' ')[0]
 const getText = line => line.replace(/^(text|placeholder) /, '')
 
+const locales = require('i18n-locales')
+const LOCAL_SCOPES = locales.map(item => item.replace(/-/g, ''))
+
 function getI18n(view, rtext) {
   const text = rtext.replace(/\r\n/g, '\n')
   const lines = text.split('\n').map(line => line.trim())
 
-  // TODO refactor so that we can extract other languages
-  const obj = {}
-
+  const obj = {};
+  const defaultLanguage = {};
   let currentBlock = null
+  let currentView = null
+  const regexList = LOCAL_SCOPES.map(shortcode => new RegExp(`^when <${shortcode}`) );
+  let isTranslation = null;
+  let textKey;
 
-  lines.forEach(line => {
+  lines.forEach((line, index) => {
     if (isBlock(line)) {
+      currentBlock = line;
     }
-
     if (isText(line)) {
       const text = getText(line)
-
-      // TODO add path to translation View/Block Name or Type/line
-      obj[line] = withoutSlot(text)
+      isTranslation = regexList.some(regex => regex.test(lines[index-1]));
+      if(!isTranslation){
```

Comment:
Because you have a condition for when `isTranslation` is true or false, you may want to invert this check (and the code inside each bit :)) so the if follows a more natural flow. Otherwise when you read this you have to think about the fact that it is negated in the first place and then follow it.

Reply:
@dariocravero, I can do this but that could also cause a little bit of confusion as `textKey` is only set when the line is *not* a translation. I know the order of operation wouldn't be affected, but it might be a little confusing when reading the code since it looks like `textKey` is used before it's set? Happy to defer to your judgement on which way around makes more sense though :)

Reply:
Mmm I hear you. Might the reason why this is confusing be because the else doesn't check for `textKey` to be set which is a requirement on it?

```
 let textKey = null;

  lines.forEach((line, index) => {
    if (isBlock(line)) {
      currentBlock = line;
      textKey = null; // <<< reset it
    }

    if (isText(line)) {
      const text = getText(line)
      const isTranslation = regexList.some(regex => regex.test(lines[index-1]));

      if (isTranslation && textKey) {  // <<< check for the key to be in first
          const langShortCode = lines[index-1].split('<')[1];
          if(!obj.hasOwnProperty(langShortCode)){
            obj[langShortCode] = {};
          }
          obj[langShortCode][`${view}/${currentBlock}/${textKey}`] = withoutSlot(text)
      } else if (!isTranslation) {
        textKey = line;
        base[`${view}/${currentBlock}/${textKey}`] = withoutSlot(text)
      }
    }
  })
```

The second part of my comment was around the fact that
```
`${view}/${currentBlock}/${textKey}`
```
could be just put into textKey instead of it being calculated twice, like:
```
 let textKey = null;

  lines.forEach((line, index) => {
    if (isBlock(line)) {
      currentBlock = line;
      textKey = null; // <<< reset it
    }

    if (isText(line)) {
      const text = getText(line)
      const isTranslation = regexList.some(regex => regex.test(lines[index-1]));

      if (isTranslation && textKey) {  // <<< check for the key to be in first
          const langShortCode = lines[index-1].split('<')[1];
          if (!obj.hasOwnProperty(langShortCode)) {
            obj[langShortCode] = {};
          }
          obj[langShortCode][textKey] = withoutSlot(text)
      } else if (!isTranslation) {
        textKey = `${view}/${currentBlock}/${textKey}`;
        base[textKey] = withoutSlot(text)
      }
    }
  })
```

you can also probably be more explicit with returns, like:

```
 let textKey = null;

  lines.forEach((line, index) => {
    if (isBlock(line)) {
      currentBlock = line;
      textKey = null; // <<< reset it
    }

    if (isText(line)) {
      const text = getText(line)
      const isTranslation = regexList.some(regex => regex.test(lines[index-1]));

      if (!isTranslation) {
        textKey = `${view}/${currentBlock}/${textKey}`;
        base[textKey] = withoutSlot(text)
        return
      }

      if (textKey) {  // <<< check for the key to be in first
          const langShortCode = lines[index-1].split('<')[1];
          if (!obj.hasOwnProperty(langShortCode)) {
            obj[langShortCode] = {};
          }
          obj[langShortCode][textKey] = withoutSlot(text)
      }
    }
  })
```

whatever you like best :)